### PR TITLE
If LICENSE is missing from the plugin path but available in the parent path when packaging, include that

### PR DESCRIPTION
**Pull request by nirvn**
@m-kuhn , @suricactus , as discussed to fix qfieldsync releases.

### CODE REVIEW

**Comment by nirvn** [comment]
Fixes https://github.com/opengisch/qgis-plugin-ci/issues/255

File: qgispluginci/release.py [hunk 1]
```diff
@@ -193,6 +194,18 @@ def create_archive(
                         continue
                     tt.add(m.name)
 
+    # add LICENSE if not already in plugin path but available in its parent
+    parent_folder_license_copied = False
+    if not Path(f"{parameters.plugin_path}/LICENSE").is_file():
+        parent_license = Path(f"{parameters.plugin_path}/../LICENSE")
+        if parent_license.is_file():
+            shutil.copy(
+                str(parent_license.resolve()), f"{parameters.plugin_path}/LICENSE"
```
Comment (Gustry) [inline qgispluginci/release.py:203]:
```suggestion
                parent_license.resolve(), f"{parameters.plugin_path}/LICENSE"
```

Minor, but `copy` accepts Pathlib object
https://docs.python.org/3/library/shutil.html#shutil.copy

**Review by Gustry — APPROVED** [review]
Nice, thanks.

Can you just add a check in the test ? Similar to https://github.com/opengisch/qgis-plugin-ci/blob/master/test/test_release.py#L198

Just that the file exists for instance, as the `LICENSE` exists in the parent folder.

File: qgispluginci/release.py (hunk 1 above)
Comment (Guts) [reply qgispluginci/release.py:203]: +1. And do you need to resolve it?

File: qgispluginci/release.py [hunk 2]
```diff
@@ -265,6 +278,9 @@ def create_archive(
                 info.compress_type = zf.compression
                 zf.writestr(info, fl)
 
+    if parent_folder_license_copied:
+        os.remove(f"{parameters.plugin_path}/LICENSE")
```
Comment (Guts) [inline qgispluginci/release.py:282]: Since we're using pathlib, may we use the `unlink` method to gain consistency?

**Comment by nirvn** [comment]
@Gustry , @Guts , comments addressed.

**Review by Gustry — APPROVED** [review]

**Comment by Gustry** [comment]
It seems the test is failing on the LICENCE file now : 

https://github.com/opengisch/qgis-plugin-ci/actions/runs/10196486393/job/28207357970?pr=307#step:7:465

For the false positive about transifex, for next time, you can push your work branch on the main git repo opengisch, it will launch the test suite without any issue in the PR.

**Review by Guts — APPROVED** [review]
LGTM. 

Thanks for fulfilling the task that I should have done.

Just for the _beauté du geste_, recently I try to work only with pathlib to deal with files and reserve shutil (or other lower-level libs) for special cases. And it's possible to use pathlib to move a file (rename) or copy one touch + write_text in oneliner.

**Comment by nirvn** [comment]
@Gustry , I'll seek to gain push permission into this repository :)

**Comment by nirvn** [comment]
@Gustry , license test failure fixed.

**Comment by nirvn** [comment]
@Gustry , @Guts , OK to merge?

**Comment by m-kuhn** [comment]
Shall we do a new release?

**Comment by Gustry** [comment]
CI is running for 2.8.9 ;-)

**Pull request merged**